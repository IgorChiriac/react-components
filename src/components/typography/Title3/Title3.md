```js
<Title3>Title3</Title3>
```

```js
<Title3 style={{ textAlign: 'center', color: 'salmon' }}>Title3</Title3>
```
