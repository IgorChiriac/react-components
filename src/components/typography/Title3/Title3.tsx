import React from 'react';
import styled from 'styled-components';
import * as fonts from '../../../constants/fonts';

export interface ITitle3Props extends React.HTMLAttributes<HTMLHeadingElement> {}

const STitle = styled.h1`
  font-weight: normal;
  font-size: 24px;
  line-height: 32px;
  letter-spacing: 0;
  font-family: ${fonts.openSans};
`;

const Title3 = (props: ITitle3Props) => {
  return <STitle {...props} />;
};

export default Title3;
