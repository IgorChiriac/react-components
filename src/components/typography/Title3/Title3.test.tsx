import { axe } from 'jest-axe';
import React from 'react';
import { render } from 'react-testing-library';
import Title3 from './Title3';

describe('<Title3 />', () => {
  it('renders the component with props with no a11y violations', async () => {
    const { container } = render(<Title3 style={{ textAlign: 'center' }}>test</Title3>);
    expect(container.firstChild).toMatchSnapshot();
    const results = await axe(container.innerHTML);
    expect(results).toHaveNoViolations();
  });
});
