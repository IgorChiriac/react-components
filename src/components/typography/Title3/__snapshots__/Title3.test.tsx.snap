// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Title3 /> renders the component with props with no a11y violations 1`] = `
.c0 {
  font-weight: normal;
  font-size: 24px;
  line-height: 32px;
  -webkit-letter-spacing: 0;
  -moz-letter-spacing: 0;
  -ms-letter-spacing: 0;
  letter-spacing: 0;
  font-family: "Open Sans",-apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol";
}

<h1
  class="c0"
  style="text-align: center;"
>
  test
</h1>
`;
